src/app/itn_orchestrator/src/keyloader.go

```diff
@@ -66,54 +65,61 @@ type KeyloaderParams struct {
 	PasswordEnv string `json:"passwordEnv,omitempty"`
 }
 
-func LoadPrivateKeyFiles(log logging.StandardLogger, params KeyloaderParams, output func(itn_json_types.MinaPrivateKey)) error {
-	entries, err := os.ReadDir(params.Dir)
+func LoadPrivateKey(fname string, password []byte) ([]byte, error) {
+	jsonFile, err := os.Open(fname)
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("failed to open keyfile %s: %v", fname, err)
+	}
+	defer jsonFile.Close()
+	bs, err := io.ReadAll(jsonFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read keyfile %s: %v", fname, err)
+	}
+	sk, err := DecodePrivateKey(bs, password)
+	if err != nil {
+		return nil, errors.Join(err, fmt.Errorf("failed decoding private key from file %s", fname))
 	}
+	return sk, nil
+}
+
+func DecodePrivateKey(bs []byte, password []byte) ([]byte, error) {
+	var box secretBox
+	json.Unmarshal(bs, &box)
+	if box.Box_primitive != "xsalsa20poly1305" || box.Pw_primitive != "argon2i" {
```

Comment:
What's those?

Reply:
I implemented parsing of our password-protected private key format in Go (it wasn't hard and is a more robust solution comparing to relying on standalone `mina` executable).

This code is there for a long time, in this PR I only move it from one larger function to a smaller one (and add a simple unit test).